src/ff/ec_prime_field.rs

```diff
@@ -266,4 +288,17 @@ mod test {
         let ia = a.invert();
         assert_eq!(a * ia, Fp25519(Scalar::ONE));
     }
+
+    /// test conversion BA256 to Fp25519
+    #[test]
+    fn test_ba256_to_fp25519() {
+        let mut rng = thread_rng();
+        let mut b = rng.gen::<BA256>();
+        b.set(255, Boolean::ZERO);
+        b.set(254, Boolean::ZERO);
+        b.set(253, Boolean::ZERO);
+        b.set(252, Boolean::ZERO);
+        b.set(251, Boolean::ZERO);
```

Comment:
This still does not make sense to me.

Isn't the prime 2^255 - 19? It would seem that you only need to zero out the highest two bits in that case to ensure that the value being expressed is less than p. Why are you zeroing out the 5 highest bits?

Reply:
This doesn't seem right to me either.  Bernstein does recommend fixing some bits to either zero or one, but as follows:

> Inside your program, to generate a 32-byte Curve25519 secret key, start by generating 32 secret random bytes from a cryptographically safe source: mysecret[0], mysecret[1], ..., mysecret[31]. Then do
>
>```
>     mysecret[0] &= 248;
>     mysecret[31] &= 127;
>     mysecret[31] |= 64;
>```

That amounts to clearing three low bits, clearing bit 256 (obviously), and setting bit 255.

Reply:
@danielmasny - why was this comment marked as resolved? I do not see any changes or updated comments or response from you.